server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -559,10 +558,24 @@ public void setCurrentSecretVersionByName(String name, long versionId, String up
    * @param name of secret series to delete.
    */
   public void deleteSecretsByName(String name) {
+    deleteSecretsByName(name, SecretDeletionMode.SOFT);
+  }
+
+  public void deleteSecretsByName(String name, SecretDeletionMode mode) {
```

Comment:
Rather than touch a zillion files with a refactor, I just left the old signature intact as an overload.